src/resqml2_0_1/PropertyKind.cpp

```diff
@@ -158,6 +158,11 @@ bool PropertyKind::isParentPartial() const
 	return parentPk->isPartial();
 }
 
+std::string PropertyKind::getUomAsString() const
+{
+	return gsoap_resqml2_0_1::soap_resqml20__ResqmlUom2s(gsoapProxy2_0_1->soap, getSpecializedGsoapProxy()->RepresentativeUom);
```

Comment:
Can be an exotic uom.
See implemention in resmql2_0_1/ContinuousProperty.cpp l.107 to support extra metadata in getUomAsString

Reply:
I updated the implementation.